### DISCUSSION TITLE
What are "ISV plug-ins"

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/eclipse.platform#765] [Q&A]
`IResource.setSessionProperty(QualifiedName, Object)` mentioned in the javadoc that

> Sessions properties are intended to be used as a caching mechanismby ISV plug-ins. 

but I can't find a good explanation what are ISV means or what are the properties of such ISV plug-ins...

**Comment by akurtakov** [comment]
ISV bundles are documentation plugins that contain generated javadoc, schema documentation and some other extra docs:
* https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/tree/master/eclipse.platform.common/bundles/org.eclipse.jdt.doc.isv
* https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/tree/master/eclipse.platform.common/bundles/org.eclipse.pde.doc.user
* https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/tree/master/eclipse.platform.common/bundles/org.eclipse.platform.doc.isv

**Reply by akurtakov** [reply]
Long story short - these are the bundles that should contain the documentation for downstream developers to read when they develop their bundles.

**Reply by laeubi** [reply]
Thats really strange as this actually seems be used for different caching purposes e.g.

```
// add the image descriptor as a session property so that it will be
// picked up by the workbench label provider upon the next update.
try {
	if (file.getSessionProperty(WorkbenchFile.IMAGE_CACHE_KEY) != image)
		file.setSessionProperty(WorkbenchFile.IMAGE_CACHE_KEY, image);
} catch (CoreException e) {
	// ignore - not being able to cache the image is not fatal
}
```

**Reply by akurtakov** [reply]
Most likely this is remnant from the very distant past and quite likely to not be helping nowadays at all.

**Reply by merks** [reply]
https://www.gartner.com/en/information-technology/glossary/isv-independent-software-vendor

So it's documentation that downstream developers use to develop stuff that exploits the framework and tools.

**Reply by Bananeweizen** [reply]
@laeubi  You've probably seen that a lot before, without realizing it. This plugin developers guide is an ISV bundle (look at the status bar):
![grafik](https://github.com/eclipse-platform/eclipse.platform/assets/406876/1c7ab78e-7486-4671-84d8-22007323ff6d)

Also the third item in each eclipse new and noteworthy webpage is named ISV, because it addresses others integrating the platform.
